klayout/drc/rule_decks/main.drc

```diff
@@ -0,0 +1,1072 @@
+        polygons_count  += count
+
+
+        metaltop_drawn = get_polygons(53 , 0 )
+        count   = metaltop_drawn.count()
+        logger.info("metaltop_drawn has %d polygons" % [count])
+        polygons_count  += count
+
+        metaltop_dummy = get_polygons(53 , 4 )
+        count   = metaltop_dummy.count()
+        logger.info("metaltop_dummy has %d polygons" % [count])
+        polygons_count  += count
+
+        metaltop       = metaltop_drawn + metaltop_dummy
+
+        metaltop_label = get_polygons(53 , 10)
+        count   = metaltop_label.count()
+        logger.info("metaltop_label has %d polygons" % [count])
+        polygons_count  += count
+
+        metaltop_slot  = get_polygons(53 , 3 )
+        count   = metaltop_slot.count()
+        logger.info("metaltop_slot has %d polygons" % [count])
+        polygons_count  += count
+
+        metalt_blk     = get_polygons(53 , 5 )
+        count   = metalt_blk.count()
+        logger.info("metalt_blk has %d polygons" % [count])
+        polygons_count  += count
+
+        top_via       = via5
+        topmin1_via   = via4
+        top_metal     = metaltop
+        topmin1_metal = metal5
+      end
+    end
+  end
+end
+
+pad            = get_polygons(37 , 0 )
+count   = pad.count()
+logger.info("pad has %d polygons" % [count])
+polygons_count  += count
+
+ubmpperi       = get_polygons(183, 0 )
+count   = ubmpperi.count()
+logger.info("ubmpperi has %d polygons" % [count])
+polygons_count  += count
+
+ubmparray      = get_polygons(184, 0 )
+count   = ubmparray.count()
+logger.info("ubmparray has %d polygons" % [count])
+polygons_count  += count
+
+ubmeplate      = get_polygons(185, 0 )
+count   = ubmeplate.count()
+logger.info("ubmeplate has %d polygons" % [count])
+polygons_count  += count
+
+metal1_res     = get_polygons(110, 11)
+count   = metal1_res.count()
+logger.info("metal1_res has %d polygons" % [count])
+polygons_count  += count
+
+metal2_res     = get_polygons(110, 12)
+count   = metal2_res.count()
+logger.info("metal2_res has %d polygons" % [count])
+polygons_count  += count
+
+metal3_res     = get_polygons(110, 13)
+count   = metal3_res.count()
+logger.info("metal3_res has %d polygons" % [count])
+polygons_count  += count
+
+metal4_res     = get_polygons(110, 14)
+count   = metal4_res.count()
+logger.info("metal4_res has %d polygons" % [count])
+polygons_count  += count
+
+metal5_res     = get_polygons(110, 15)
+count   = metal5_res.count()
+logger.info("metal5_res has %d polygons" % [count])
+polygons_count  += count
+
+metal6_res     = get_polygons(110, 16)
+count   = metal6_res.count()
+logger.info("metal6_res has %d polygons" % [count])
+polygons_count  += count
+
+pr_bndry       = get_polygons(0  , 0 )
+count   = pr_bndry.count()
+logger.info("pr_bndry has %d polygons" % [count])
+polygons_count  += count
+
+border         = get_polygons(63 , 0 )
+count   = border.count()
+logger.info("border has %d polygons" % [count])
+polygons_count  += count
```

Comment:
is there a way to get a total polygon count directly from the layout (without summing up manually) (i.e: we might forget a layer)

Reply:
I believe that's a question to @klayoutmatthias

Reply:
something like this?
```
logger.info("Total no. of polygons: #{ layers.map { |l| polygons(l).count() }.sum }")
```
and it currently display an higher level of polygon count for me (`1053661`) than the currently `polygons_count` logic (`891613`) for the caravel_user_project's `user_proj_example` gds for gf180mcu.

Reply:
added an issue #18